demos/sample_azure_iot_pnp/sample_azure_iot_pnp_data_if.h

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+/**
+ * @brief Defines an interface to be used by samples when interacting with sample_azure_iot_pnp.c module.
+ *        This interface allows the module implementing the specific plug-and-play model to exchange data
+ *        with the module responsible for communicating with Azure IoT Hub.
+ */
+
+#ifndef SAMPLE_AZURE_IOT_PNP_H
+#define SAMPLE_AZURE_IOT_PNP_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "azure_iot_hub_client_properties.h"
+
+/**
```

Comment:
Since it depends on the sampleazureiotMODEL_ID, that would make sense. Otherwise it would need to follow a specific order of includes in sample_azure_iot_pnp.c

Reply:
Done.

Reply:
Done